bin/all_sky_search/pycbc_fit_sngls_over_multiparam

```diff
@@ -97,13 +101,11 @@ def smooth_n_closest(nabove, invalphan, ntotal, dists, total_trigs=500):
     templates_required = 0
     n_triggers = 0
     # Count number of templates required to gather total_trigs templates,
-    # start at closest
-    # total_trigs, if supplied on command line will be a str so convert to int
-    while n_triggers < int(total_trigs):
-        n_triggers += nabove[dist_sort[n_triggers]]
-        templates_required += 1
-    logging.debug("%d templates required to obtain %d triggers",
-                  templates_required, n_triggers)
+    # starting at closest
+    ntcs = nabove[dist_sort].cumsum()
+    templates_required = numpy.searchsorted(ntcs, total_trigs) + 1
```

Comment:
Yes that's it. The problem before was that `dist_sort[n_triggers]` was basically nonsense, as it was indexing on the number of triggers rather than the template id. (No idea how it didn't error out though...).

The solution removes the manual cumulative sum and then uses searchsorted rather than a manual search.